### PR TITLE
Associated companies fixes

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: packages/common/.eslintrc.cjs [hunk 1]
```diff
@@ -1,4 +1,7 @@
 /** @type {import('eslint').Linter.Config} */
 module.exports = {
   extends: ['@ballerine/eslint-config'],
+  parserOptions: {
+    project: './tsconfig.eslint.json',
+  },
```
Comment (Omri-Levy) [inline packages/common/.eslintrc.cjs]: Boyscout ESLint fix.

File: packages/common/src/utils/zod-builder/zod-builder.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { ZodTypeAny } from 'zod';
+
+export const zodBuilder =
+  <TType>() =>
+  <TSchema extends ZodTypeAny & { _output: TType }>(schema: TSchema) =>
+    schema;
```
Comment (Omri-Levy) [inline packages/common/src/utils/zod-builder/zod-builder.ts:6]: Ensures a zod schema follows a passed TypeScript type.

File: packages/workflow-core/.eslintrc.cjs [hunk 3]
```diff
@@ -1,4 +1,7 @@
 /** @type {import('eslint').Linter.Config} */
 module.exports = {
   extends: ['@ballerine/eslint-config'],
+  parserOptions: {
+    project: './tsconfig.eslint.json',
+  },
```
Comment (Omri-Levy) [inline packages/workflow-core/.eslintrc.cjs:6]: Boyscout ESLint fix.

File: packages/workflow-core/src/lib/plugins/common-plugin/iterative-plugin.ts [hunk 4]
```diff
@@ -51,11 +52,14 @@ export class IterativePlugin {
...
       mutatedRecord = await this.transformByTransformer(transformer, mutatedRecord);
     }
+
     return mutatedRecord;
   }
 
   async transformByTransformer(transformer: Transformer, record: AnyRecord) {
-    console.log(`transformByTransformer() called for mapping: ${transformer.mapping}`);
+    console.log(
+      `transformByTransformer() called for mapping: ${JSON.stringify(transformer.mapping)}`,
```
Comment (Omri-Levy) [inline packages/workflow-core/src/lib/plugins/common-plugin/iterative-plugin.ts:61]: Boyscout. Previously logged out `[object Object]`.

File: services/workflows-service/src/filter/dtos/temp-zod-schemas.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ export const zQueryModeEnum = z.enum(['Default', 'Insensitive']);
 export const zEnumerable = <TSchema>(schema: ZodSchema<TSchema>) =>
   z.union([schema, schema.array()]);
 
-// @ts-expect-error - It is expected for z.lazy to be any.
+// @ts-ignore - It is expected for z.lazy to be any.
```
Comment (Omri-Levy) [inline services/workflows-service/src/filter/dtos/temp-zod-schemas.ts:15]: Jest fails when using `expect-error`.

File: packages/common/src/utils/zod-builder/zod-builder.ts (hunk 2 above)
Comment (liorzam) [inline packages/common/src/utils/zod-builder/zod-builder.ts:6]: I would like to have an explanation about it....

i already would say that lets try to use [ajv-ts](https://github.com/vitalics/ajv-ts) instead keep working on advanced zod schema builders

File: services/workflows-service/src/business/schemas.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    endUsersOnBusinesses: z.array(RecordAnySchema).optional(),
+  }),
+);
+
+export const BaseBusinessCreateInputSchema = zodBuilder<
+  Omit<Prisma.BusinessCreateInput, 'project' | 'endUsers' | 'endUsersOnBusinesses'>
+>()(
+  z.object({
+    id: z.string().cuid().optional(),
+    correlationId: z.string().optional(),
```
Comment (liorzam) [inline services/workflows-service/src/business/schemas.ts:23]: @MatanYadaev Did we say its gonna be required field?

Comment (MatanYadaev) [reply services/workflows-service/src/business/schemas.ts:23]: We concluded it, yes, but who's going to refactor it? :)

Comment (Omri-Levy) [reply services/workflows-service/src/business/schemas.ts:23]: Out of scope for my task. I'll update this schema myself once the field is no longer optional.

**Review by tomer-shvadron — APPROVED** [review]

**Pull request merged**